src/dex-ui/pages/CreateProposal/ProposalForms/ContractUpgradeProposalForm.tsx

```diff
@@ -0,0 +1,190 @@
+import { FormControl, Input, FormErrorMessage, Flex, Button, Spacer, Text, Box } from "@chakra-ui/react";
+import { WarningIcon } from "@chakra-ui/icons";
+import { ReactElement } from "react";
+import { useForm, Controller } from "react-hook-form";
+import { useNavigate, Link as ReachLink } from "react-router-dom";
+import { LoadingDialog, TextEditor, Breadcrumb, ArrowLeftIcon } from "@dex-ui-components";
+import { isValidUrl } from "@utils";
+import { ContractUpgradeProposalFormData, CreateProposalLocationProps } from "./types";
+import { TransactionResponse } from "@hashgraph/sdk";
+import { useCreateContractUpgradeProposal } from "@dex-ui/hooks";
+import { Paths } from "@dex-ui/routes";
+import { Page, PageHeader } from "@dex-ui/layouts";
+
+export function ContractUpgradeProposalForm(): ReactElement {
+  const navigate = useNavigate();
+  const {
+    handleSubmit,
+    register,
+    control,
+    reset,
+    getValues,
+    formState: { errors },
+  } = useForm<ContractUpgradeProposalFormData>();
+
+  const createContractUpgradeMutationResults = useCreateContractUpgradeProposal(handleSendProposesSuccess);
+  const {
+    isLoading,
+    isError,
+    error,
+    mutate,
+    reset: resetCreateContractUpgradeTransaction,
+  } = createContractUpgradeMutationResults;
+
+  function handleSendProposesSuccess(transactionResponse: TransactionResponse) {
+    const createProposalLocationProps = {
+      state: {
+        proposalTitle: getValues().title,
+        proposalTransactionId: transactionResponse?.transactionId.toString(),
+        isProposalCreationSuccessful: true,
+      } as CreateProposalLocationProps,
+    };
+    reset();
+    navigate(Paths.Governance.absolute, createProposalLocationProps);
+  }
+
+  function handleCancelClick() {
+    reset();
+    navigate(Paths.Governance.absolute);
+  }
+
+  async function onSubmit(data: ContractUpgradeProposalFormData) {
+    mutate({
+      title: data.title,
+      description: data.description,
+      linkToDiscussion: data.linkToDiscussion,
+      newContractProxyId: data.newContractProxyId,
+      contractToUpgrade: data.contractToUpgrade,
+    });
+  }
+
+  return (
+    <Page
+      header={
+        <PageHeader
+          leftContent={[<Text textStyle="h2">Create New Proposal</Text>]}
+          rightContent={[
+            <Breadcrumb
+              to={`${Paths.Governance.absolute}/${Paths.Governance.SelectProposalType}`}
+              as={ReachLink}
+              label="Back to Select Proposal Type"
+              leftIcon={<ArrowLeftIcon />}
+            />,
+          ]}
+        />
+      }
+      body={
+        <>
+          <form onSubmit={handleSubmit(onSubmit)}>
```

Comment:
Note for the future: I'm working on a FormLayout component to wrap all of these inline forms. This will make it easier to create, style, and maintain these forms/wizards.

Reply:
Great.!!